### PR TITLE
quick-serve: init at 0.2.0

### PR DESCRIPTION
**Pull request by vinnymeller**
## Description of changes

Adds the [quick-serve](https://github.com/joaofl/quick-serve) application

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by vinnymeller** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>quick-serve</li>
  </ul>
</details>

**Comment by vinnymeller** [comment]
Redid after force push:

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>quick-serve</li>
  </ul>
</details>

File: pkgs/by-name/qu/quick-serve/package.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  };
+
+  cargoHash = "sha256-Cogt3eS97CM0Ym0k1C74NfIzNeC3ajIWdTsvwZtRL0U=";
+
+  patches = [./remove-e2e-test-path.patch];
+
+  nativeBuildInputs = [
+    pkg-config
+    autoPatchelfHook
+    wget
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-serve/package.nix:36]: Why do we need wget here? Downloading things doesn't work inside the sandbox.

File: pkgs/by-name/qu/quick-serve/package.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  buildInputs =
+    [
+      libxkbcommon
+      vulkan-loader
+      zstd
+      gtk3
+      libGL
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-serve/package.nix:49]: Please use with here

File: pkgs/by-name/qu/quick-serve/package.nix [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    hash = "sha256-yO2D3w+aURLX8caFuadzRPdQDlI37ZB4CFzvO4J/zq8=";
+  };
+
+  cargoHash = "sha256-Cogt3eS97CM0Ym0k1C74NfIzNeC3ajIWdTsvwZtRL0U=";
+
+  patches = [./remove-e2e-test-path.patch];
+
+  nativeBuildInputs = [
+    pkg-config
+    autoPatchelfHook
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-serve/package.nix:35]: Please set [source provenance](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#source-provenance-sec-meta-sourceprovenance) when using recompiled binaries.

File: pkgs/by-name/qu/quick-serve/package.nix (hunk 1 above)
Comment (vinnymeller) [reply pkgs/by-name/qu/quick-serve/package.nix:36]: it seems to work for localhost? it's for this test & a couple similar: https://github.com/joaofl/quick-serve/blob/0c654a9ecbff027779dd03565b4f0d284f0c4dce/src/servers/http.rs#L83-L92 checking that the various servers run

Comment (SuperSandro2000) [reply pkgs/by-name/qu/quick-serve/package.nix:39]: If it is for tests, then it should go into nativeCheckInputs

Comment (vinnymeller) [reply pkgs/by-name/qu/quick-serve/package.nix:42]: ah sorry missed that somehow. will have to go back through some of the readmes. fixed now

File: pkgs/by-name/qu/quick-serve/package.nix [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  postInstall = ''
+    wrapProgram $out/bin/quick-serve \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDependencies}"
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Quick Serve";
+      exec = meta.mainProgram;
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-serve/package.nix:86]:
```suggestion
      exec = "quick-serve";
```

File: pkgs/by-name/qu/quick-serve/package.nix [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      desktopName = "Quick Serve";
+      comment = meta.description;
+      categories = ["Network" "Utility"];
+    })
+  ];
+
+  meta = with lib; {
+    description = "A zero-config, multi-platform, multi-protocol standalone server for a prompt file serving";
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-serve/package.nix:94]:
```suggestion
    description = "Zero-config, multi-platform, multi-protocol standalone server for a prompt file serving";
```

File: pkgs/by-name/qu/quick-serve/package.nix [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    wrapProgram $out/bin/quick-serve \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDependencies}"
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Quick Serve";
+      exec = meta.mainProgram;
+      desktopName = "Quick Serve";
+      comment = meta.description;
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-serve/package.nix:88]:
```suggestion
      comment = "Standalone server for a prompt file serving";
```

**Pull request closed without merging**